debezium-connector-cassandra/src/main/java/io/debezium/connector/cassandra/CommitLogReadHandlerImpl.java

```diff
@@ -395,23 +395,30 @@ private void handleRowModifications(Row row, RowType rowType, PartitionUpdate pu
     private void populatePartitionColumns(RowData after, PartitionUpdate pu) {
         List<Object> partitionKeys = getPartitionKeys(pu);
         for (ColumnDefinition cd : pu.metadata().partitionKeyColumns()) {
-            String name = cd.name.toString();
-            Object value = partitionKeys.get(cd.position());
-            CellData cellData = new CellData(name, value, null, CellData.ColumnType.PARTITION);
-            after.addCell(cellData);
+            try {
+                String name = cd.name.toString();
+                Object value = partitionKeys.get(cd.position());
+                CellData cellData = new CellData(name, value, null, CellData.ColumnType.PARTITION);
+                after.addCell(cellData);
+            }
+            catch (Exception e) {
+                LOGGER.error("Failed to populate Column {} with Type {} of Table {} in KeySpace {}.",
```

Comment:
Let's better either log _or_ rethrow, but not both at the same time, as it'll clutter log files. So I'd suggest to wrap the original exception in a `DebeziumException`, adding the message with the additional context you'd like to propagate. It'll all be logged at once then.